rag/flow/chunker/title_chunker/hierarchy_chunker.py

```
@@ -95,16 +98,21 @@ def flush_text_records():
-                    )
-                    if path
-                )
+                for path in root.get_paths(
+                    target_level,
+                    self.param.include_heading_content,
+                ):
+                    if not path:
+                        continue
+                    if any(index >= context_boundary for index in path):
+                        record_groups.append([text_records[index] for index in path])
+            heading_records = [rec for rec, lvl in zip(text_records, text_levels) if 0 < lvl < BODY_LEVEL]
```

Ruff (0.16.1)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)